### ISSUE TITLE
[Sponge] Issue with permission checks on User object upon repeated login

### ISSUE THREAD

**Issue by dualspiral** [lucko/LuckPerms#108]
```
Minecraft Version: 1.10.2
Sponge Version: SpongeForge 1.10.2-2171-5.1.0-BETA-2008
Sponge API Version: SpongeAPI 5.1.0-SNAPSHOT-323db46
LuckPerms v2.16.60.
```

There is a strange issue occuring where a permission check on a `Player` object and the same check on their corresponding `User` object can return different results. This happens on second and subsequent player logins.

To reproduce what happens, I gave myself the permission `test.test` and wrote a plugin that does a permission check on the Player object, then the User object. I then restarted my server, logged in, ran `/test`, noted the console output, logged out, logged in again and ran `/test`. In both cases, the check on the player object (`Player#hasPermission(String)`) returned correctly, but the check on the user object (`User#hasPermission(String)`) only returns correctly the first time. I would expect that the call to `User#hasPermission(S)` returns the same result as `Player#hasPermission(S)` _every_ time.

Performing the check later will return the correct result, so I'd guess there is some weird caching thing going on, but you know the code better than me!

The code and results are in the following gist: https://gist.github.com/dualspiral/08bf3244fc04e7402f2d2feb1968d374

**Comment by lucko** [comment]
Would you mind testing with Sponge PEX? Sounds more like an issue with the way Sponge caches it here: https://github.com/SpongePowered/SpongeCommon/blob/bleeding/src/main/java/org/spongepowered/common/mixin/core/command/MixinSubject.java

**Comment by dualspiral** [comment]
It works fine in PEX.

https://gist.github.com/dualspiral/cbd1a116b8812d4161c3e393a7687b5c

**Comment by dualspiral** [comment]
Just looking into it, you're probably right that it's a Sponge thing, but then, I'm not sure why PEX works and LP doesn't. I'll probably test it with PermissionManager at some point tomorrow (I need to test something with that anyway), I'll keep you in the loop as to what I find, if anything.

I've worked around the issue in Nucleus for now - just forced the check on `Player` if it exists, and I plan to get that update out tomorrow, so don't feel you need to rush this for my sake!

**Comment by bloodmc** [comment]
@lucko @dualspiral I just experienced this myself testing GP with latest LP as I'm trying to cache some perm checks for performance sake. Upon logging in, the user subject always returns false. However, the odd part is, if I add a hasPermission check during player login then the user subject seems to always work. This tells me that the user subject's data is not properly initialized unless an actual player logs in and runs a permission check. I performed the same test with PEX and had no issues.
